src/common/DOMWorld.ts

```diff
@@ -482,19 +489,23 @@ export class DOMWorld {
   /**
    * @internal
    */
-  async addBindingToContext(name: string) {
+  async addBindingToContext(
+    context: ExecutionContext,
+    name: string
+  ): Promise<void> {
     // Previous operation added the binding so we are done.
-    if (this._ctxBindings.has(name)) return;
+    if (this._ctxBindings.has(name + context._contextId)) {
```

Comment:
Instead of just keeping track of the names we have set up bindings for, we remember `binding + contextId`.
At least theoretically, I think we could otherwise end up in a situation where we added the binding just as the context was changed. If so, we could end up putting the name in `_ctxBindings` even though it isn't added in the new context.

Reply:
Currently, we get the same key for:

```
name = 'abc'
contextId = 'def'
```

and 

```
name = 'ab'
contextId = 'cdef'
```

...which seems potentially bad. I think we should we do something like

```
`${name}_${context._contextId}`
```

That is, always separate `name` and `contextId` by _something_ (not necessarily `_` — ideally some character that cannot occur in `contextId`s) to avoid collisions between overlapping characters.

Reply:
Good idea! I've added a `_` separator. The `contextId`s are numbers, so `_` should do.